pkg/operator/oauthapiencryptioncontroller/oauthapiencryptioncontroller.go

```diff
@@ -0,0 +1,128 @@
+//         - it will add the secret with the annotation (must be atomic operation) because CAO will start managing its own config iff the secret exist without the annotation
+//         - it will start synchronisation
+//
+// case 2: if the secret exists and it is annotated
+//         - it will simply start synchronisation
+//
+// case 3: no-op: when the secret exits but it doesn't have the annotation - that means it was created by CAO in 4.6 and this is downgrade
+// case 4: no-op: when the secret doesn't exist and encryption is off
+//
+// drawbacks:
+// - it will not recover when the annotation was manually removed by a user,
+//   to recover we would have to put a value in the annotation instead and coordinate OAS-A and CAO but then we would have to remember to add it in CAO (4.6) as well
+func (c *oauthAPIServerController) sync(ctx context.Context, controllerContext factory.SyncContext) error {
+	openshiftAPIServerEncryptionCfg, err := c.secretLister.Get(fmt.Sprintf("%s-%s", encryptionconfig.EncryptionConfSecretName, c.targetNamespace))
+	if apierrors.IsNotFound(err) {
+		return nil // case 4: encryption off
+	}
+	if err != nil {
+		return err
+	}
+
+	oauthAPIServerEncryptionCfg, err := c.secretLister.Get(fmt.Sprintf("%s-%s", encryptionconfig.EncryptionConfSecretName, c.oauthAPIServerTargetNamespace))
+	if err != nil && !apierrors.IsNotFound(err) {
+		return err
+	}
+	if apierrors.IsNotFound(err) {
+		// case 1: create with annotation and start synchronization
+		source := openshiftAPIServerEncryptionCfg.DeepCopy()
+		source.Name = fmt.Sprintf("%s-%s", encryptionconfig.EncryptionConfSecretName, c.oauthAPIServerTargetNamespace)
+		source.ResourceVersion = ""
+		source.OwnerReferences = []metav1.OwnerReference{}
+		if source.Annotations == nil {
+			source.Annotations = map[string]string{}
+		}
+		source.Annotations[encryptionConfigManagedBy] = encryptionConfigManagedByValue
+		if _, _, err = resourceapply.ApplySecret(c.secretClient, controllerContext.Recorder(), source); err != nil {
+			return err
+		}
+		return c.startSecretSynchronization(controllerContext.Recorder())
+	}
+	if _, exist := oauthAPIServerEncryptionCfg.Annotations[encryptionConfigManagedBy]; exist {
+		// case 2: exists and it is annotated
+		return c.startSecretSynchronization(controllerContext.Recorder())
+	}
+
+	// case 3: no-op the secret is managed by CAO
+	return nil
+}
+
+func (c *oauthAPIServerController) startSecretSynchronization(recorder events.Recorder) error {
+	if c.synchronizationEnabled {
+		return nil
+	}
+
+	if err := c.resourceSyncer.SyncSecret(
+		resourcesynccontroller.ResourceLocation{Namespace: operatorclient.GlobalMachineSpecifiedConfigNamespace, Name: fmt.Sprintf("%s-%s", encryptionconfig.EncryptionConfSecretName, c.oauthAPIServerTargetNamespace)},
+		resourcesynccontroller.ResourceLocation{Namespace: operatorclient.GlobalMachineSpecifiedConfigNamespace, Name: fmt.Sprintf("%s-%s", encryptionconfig.EncryptionConfSecretName, c.targetNamespace)},
```

Comment:
also here, better use client primitives. It's easier to reason about than these with less clear semantics.

Reply:
`resourceSyncer.SyncSecret` will keep two secrets in `sync`

Reply:
this is a one-time call?

Reply:
i.e. no start of a controller. Then maybe it's fine.

Reply:
yes, this is a one-time call, protected by `c.synchronizationEnabled`

Reply:
How do we stop it again?

Reply:
we don't, In `4.6` we will remove the code, on downgrades (the secret exist but doesn't have the annotation) we do nothing.